zebra-chain/src/parameters/network.rs

```diff
@@ -173,7 +277,7 @@ impl FromStr for Network {
     fn from_str(string: &str) -> Result<Self, Self::Err> {
         match string.to_lowercase().as_str() {
             "mainnet" => Ok(Network::Mainnet),
-            "testnet" => Ok(Network::Testnet),
+            "testnet" => Ok(Network::new_default_testnet()),
```

Comment:
Consider reserving "Mainnet", "Testnet", and "Regtest" network names (panicking if a configured testnet uses them) so we don't return a default testnet here for configured testnets.

TODO: Check where this conversion is used.